doc/modules/base-libraries/pages/stdlib-intro.adoc

```diff
@@ -33,13 +33,10 @@ import Types "./types";
 ....
 
 [[compiler-prelude]]
-== Using predefined compiler prelude function
 
-The compiler itself contains a *prelude* in every program, effectively extending the language definition with a set of types and declarations that are always available.
+== The compiler primitives and prelude
 
-In addition to these built-in compiler definitions, the prelude of the base library contributes some additional functions in a supplemental Prelude module.
-
-In contrast to the definitions in other modules, the compiler prelude definitions are always present, and do _not_ require any `import` declaration to bring them into the current scope.
+There are two modules that have a slightly special status in a {proglang} program, since they are intrinsically part of the compiler, as opposed to originating from a library that is  presented as source code. The primitives are usually imported as `import Prim "mo:prim"` and contain special-purpose routines dealing with built-in data types. The compiler prelude provides higher-level abstractions and helpers that serve as a bridge between {proglang}’s surface syntax and the compiler’s inner workings. The prelude is regularly imported as `import P "Prelude"`.
```

Comment:
Quite seriously, I think we need to remove this section altogether. Both these modules are implementation details of either the compiler or the base library. Normal programs should _never_ use them! It's problematic enough that they still can, we should certainly not encourage that by blessing them in the docs.

Reply:
Fully agreed that `Prim` need not be covered here.

The `Prelude` in the motoko-base libraray has _nothing_ to do with the prelude in the compiler. The prelude in the compiler needs not be discussed here. And I don't know if we need the `Prelude` module in motoko-base; it seems to be an odd catch-all. Maybe rename to `Util` or put its content into more appropriate other modules.

And then yes, this section can go.

Reply:
Yay, gone! Now I see that `src/Prelude.mo` is an internal module, not linked to the compiler.